src/dotnet/APIView/APIView/Languages/CodeFileBuilder.cs

```diff
@@ -511,15 +511,15 @@ private CodeFileToken MapToken(ISymbol definedSymbol, SymbolDisplayPart symbolDi
             var symbol = symbolDisplayPart.Symbol;
 
             if (symbol is INamedTypeSymbol &&
-                (definedSymbol == null || !definedSymbol.Equals(symbol)) &&
-                _assembly.Equals(symbol.ContainingAssembly))
+                (definedSymbol == null || !SymbolEqualityComparer.Default.Equals(definedSymbol, symbol)) &&
+                SymbolEqualityComparer.Default.Equals(_assembly, symbol.ContainingAssembly))
             {
                 navigateToId = symbol.GetId();
             }
 
             return new CodeFileToken()
             {
-                DefinitionId = definedSymbol?.Equals(symbol) == true ? definedSymbol.GetId() : null,
+                DefinitionId = (definedSymbol != null && SymbolEqualityComparer.Default.Equals(definedSymbol, symbol)) ? definedSymbol.GetId() : null,
```

Comment:
You don't need null check for `definedSymbol` here, `SymbolEqualityComparer` does it:
https://github.com/dotnet/roslyn/blob/e704ca635bd6de70a0250e34c4567c7a28fa9f6d/src/Compilers/Core/Portable/Symbols/SymbolEqualityComparer.cs#L46

Reply:
It's important for the methods logic. `SymbolEqualityComparer.Default.Equals(null, null)` returns true and we don't want it.

Reply:
Is it possible that `symbolDisplayPart.Symbol == null`?

Reply:
Yes

Reply:
Ok